src/trackers/classes/request.js

```diff
@@ -16,6 +19,17 @@ class Request {
         this.originalSubdomain = undefined
         this.responseHash = reqData.responseBodyHash
         this.nameservers = []
+        this.firstPartyCookies = site.thirdPartyJSCookies
+            .filter(cookie => cookie.source === reqData.url && // cookie source is this request
+                cookie.value && // cookie has a truthy value
+                isFirstPartyCookie(cookie.domain, site.domain)) // cookie was set on the first party origin
+        this.firstPartyCookiesSent = site.thirdPartyJSCookies
+            .filter(cookie => {
+                // only consider cookies 6 or more characters long
+                return cookie.value &&
+                    cookie.value.length > COOKIE_LENGTH_CUTOFF &&
+                    isCookieValueInUrl(cookie, ParsedUrl.parse(reqData.url)) // TODO fix with url parsing update
```

Comment:
is this TODO still needed?

Reply:
Yes. I'll tackle this as LHF before the end of the week. The issue here is we have to parse the URL twice, as there is a `ParsedUrl` type as well as the standard `URL` being used.